### ISSUE TITLE
no-angular-mock rule "module" is not defined

### ISSUE THREAD

**Issue by indiesquidge** [Gillespie59/eslint-plugin-angular#330]
Hello there. I appreciate you helping create/maintain this eslint plugin :)

However, I do not seem to be able to get some things working correctly. 

I have globally installed both `eslint` and `eslint-plugin-angular` and have successfully included the plugin in my `.eslintrc`. What I seem to be having difficulty with is using [no-angular-mock](https://github.com/Gillespie59/eslint-plugin-angular/blob/master/docs/no-angular-mock.md).

I have my spec file

```javascript
/*eslint angular/no-angular-mock: 2*/

'use strict';

describe('app.brief.giCheckboxOptional directive', function () {
  beforeEach(module('app.brief'));
});
```

However, when I check the file against `eslint`, I get the following

```sh
  6:14  error  "module" is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

I have included `/*eslint angular/no-angular-mock: 2*/` as suggested [here](https://github.com/Gillespie59/eslint-plugin-angular/blob/master/docs/no-angular-mock.md) but it does not seem to get rid of the errors. Any help would be appreciated :)

**Comment by tilmanpotthof** [comment]
The problem is not the rule `angular/no-angular-mock`, but the eslint rule `no-undef`. It prevents you from using undefined global variables, and does not know the global methods by angular-mocks (http://eslint.org/docs/rules/no-undef.html).

Quickfix: Add `module` and `inject` to your eslint globals.

    "globals": {
      "angular": false,
      "module": false,
      "inject": false
    }

@remcohaszing @Gillespie59: I think we should document this problem in the `no-angular-mock` rule, but I think it is not possible to fix this problem in general. Or is it possible to add globals for a plugin?

**Comment by Gillespie59** [comment]
I do not think it is possible to add globals for a specific plugin. Maybe a question we can ask to the ESLint project. 
But yes we can document this problem. Not in the `no-angular-mock` rule, because this error is not linked to this rule. It is a global issue. Why not creating a block in the README next to the `Default ESLint configuration file`  block ?

**Comment by indiesquidge** [comment]
@tilmanpotthof: ah I see. So as far as you guys know, there is no solution to this kind of issue other than simply adding to my globals list? Might I ask what `/*eslint angular/no-angular-mock: 2*/` is meant to do if that is the case? I think I am confused about what the [documentation](https://github.com/Gillespie59/eslint-plugin-angular/blob/master/docs/no-angular-mock.md) means.

**Comment by Gillespie59** [comment]
according to the doc, you should use, for example, `inject` instead of `angular.mock.inject`
But the fact we use `inject`, the `no-undef` rule throw an error because he do not any global variable named `inject`. 
So you are right @tilmanpotthof , mays add a comment in this rule, in order to explain to the user, that he should add all methods of the angular.mock object in his globals properties.

**Comment by indiesquidge** [comment]
Makes sense. Thank you for explaining and helping out, @Gillespie59 and @tilmanpotthof. I would be happy to create a PR to add this information to the rule documentation if you'd like :)

**Comment by Gillespie59** [comment]
Of course ;)
